engine/src/main/java/org/terasology/particles/ParticleSystemManagerImpl.java

```diff
@@ -107,11 +111,16 @@ public void update(float delta) {
     @Override
     public Stream<ParticleRenderingData> getParticleEmittersByDataComponent(Class<? extends Component> particleDataComponent) {
         return particleUpdater.getParticleEmitters().stream()
-                .filter(p -> p.ownerEntity.hasComponent(particleDataComponent))
-                .map(particleEmitterComponent ->
-                        new ParticleRenderingData<>(
-                                particleEmitterComponent.ownerEntity.getComponent(particleDataComponent),
-                                particleEmitterComponent.particlePool
-                        ));
+                .filter(emitter -> emitter.ownerEntity.hasComponent(particleDataComponent))
+                .filter(distinctByKey(emitter -> emitter.particlePool))
+                .map(emitter -> new ParticleRenderingData<>(
+                        emitter.ownerEntity.getComponent(particleDataComponent),
+                        emitter.particlePool
+                ));
+    }
+
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor) {
+        Set<Object> seen = ConcurrentHashMap.newKeySet();
+        return element -> seen.add(keyExtractor.apply(element));
```

Comment:
Took me a few minutes to understand what is happening here... In the end, pretty elegant solution.